src/Plugin/resource/Field/ResourceFieldEntity.php

```diff
@@ -201,14 +203,34 @@ protected function propertyIdentifier(\EntityMetadataWrapper $property_wrapper)
     if ($property_wrapper instanceof \EntityDrupalWrapper) {
       // The property wrapper is a reference to another entity get the entity
       // ID.
-      $embedded_identifier = $property_wrapper->getIdentifier();
-    }
-    else {
-      // The property is a regular one, get the value out of it and use it as
-      // the embedded identifier.
-      $embedded_identifier = $this->fieldValue($property_wrapper);
+      $identifier = $property_wrapper->getIdentifier() ?: NULL;
+      $resource = $this->getResource();
+      // TODO: Make sure we still want to support full_view.
+      if (!$resource || !$identifier || (isset($resource['full_view']) && $resource['full_view'] === FALSE)) {
+        return $identifier;
+      }
+      // If there is a resource that we are pointing to, we need to use the id
+      // field that that particular resource has in its configuration. Trying to
+      // load by the entity id in that scenario will lead to a 404.
+      // We'll load the plugin to get the idField configuration.
+      $instance_id = sprintf('%s:%d.%d', $resource['name'], $resource['majorVersion'], $resource['minorVersion']);
+      $plugin_manager = ResourcePluginManager::create('cache', Request::create('', array(), RequestInterface::METHOD_GET));
+      /* @var ResourceInterface $resource */
+      $resource = $plugin_manager->createInstance($instance_id);
+      $plugin_definition = $resource->getPluginDefinition();
+      if (empty($plugin_definition['dataProvider']['idField'])) {
+        return $identifier;
+      }
+      try {
+        return $property_wrapper->{$plugin_definition['dataProvider']['idField']}->value();
+      }
+      catch (\EntityMetadataWrapperException $e) {
```

Comment:
If that property does not exist in the specified entity (or is not declared to Property API)